test/config-luatest/schema_test.lua

```diff
@@ -2680,6 +2681,132 @@ g.test_map_map_itself = function()
     })
 end
 
+local function apply_vars_f(data, w, vars)
+    if data ~= nil and type(data) == 'string' then
+        return (data:gsub('{{ *(.-) *}}', function(var_name)
+            if vars[var_name] ~= nil then
+                return vars[var_name]
+            end
+            w.error(('Unknown variable %q'):format(var_name))
+        end))
+    end
+    return data
+end
```

Comment:
Nit: It is almost same as `apply_vars` above (in the `Testing helpers for <schema object>:map()` section). Can we reuse it?

Even if not, I would move this helper function to the `Testing helpers for <schema object>:map()` section.